src/controllers/roomSocket.ts

```
@@ -31,48 +44,43 @@ function createRoom(socket: Socket) {
 
-                    socket.to(data.code).emit('rooms:events', updatedRoom); // Send the room data to the other players
-                    console.log(socket.id + ' joined room ' + data.code);
+        await clientDB.collection('rooms').updateOne({code: data.code}, roleData);
 
-                    socket.to(updatedRoom.creator).emit('trapper:join', {player: socket.id}); // Send the player id to the creator
-                }
-            });
-        } else {
-            socket.emit('rooms:join', {error: 'Room not found'});
+        const updatedRoom = await clientDB.collection('rooms').findOne({code: data.code});
+        if (updatedRoom && updatedRoom.creator) {
```

Biome

[error] Change to an optional chain.

Unsafe fix: Change to an optional chain.


(lint/complexity/useOptionalChain)